test/cli/case-blank-when-body/test.out

```diff
@@ -0,0 +1,18 @@
+test/cli/case-blank-when-body/blank.rb:14: Expected `String` but found `T.nilable(String)` for method result type https://srb.help/7005
+    14 |end
+        ^^^
+  Expected `String` for result type of method `foo`:
+    test/cli/case-blank-when-body/blank.rb:6:
+     6 |def foo(x)
+        ^^^^^^^^^^
+  Got `T.nilable(String)` originating from:
+    test/cli/case-blank-when-body/blank.rb:8:
+     8 |  when 1
+     9 |  when 2
```

Comment:
This is showing the `"when 1\n"` as the location where the `nil` comes from.

That's a bit weird—that means for example that if you hovered over the `when` keyword, Sorbet would show you a revealed type of `nil`. It also means that certain autocorrects might attempt to use wrap `"when 1\n"` in a `T.must` or `T.unsafe`, which would be a syntax error.

I think that we should instead set the loc for the `MK::Nil` to `when->loc.copyEndWithZeroLength()`.

Doing this exposes another bug, which is that the location includes the `"\n"`. We should probably change the loc for the `parser::When` node that we create in Builder.cc to not include the loc of the `then` token if it's actually backed by a `tNL` token instead of a `kTHEN` token.